ledger-tool/src/program.rs

```diff
@@ -131,18 +131,52 @@ fn load_blockstore(ledger_path: &Path, arg_matches: &ArgMatches<'_>) -> Arc<Bank
     bank
 }
 
-pub trait RunSubCommand {
-    fn run_subcommand(self) -> Self;
+pub trait ProgramSubCommand {
+    fn program_subcommand(self) -> Self;
 }
 
-impl RunSubCommand for App<'_, '_> {
-    fn run_subcommand(self) -> Self {
+impl ProgramSubCommand for App<'_, '_> {
+    fn program_subcommand(self) -> Self {
+        let program_arg = Arg::with_name("PROGRAM")
+            .help(
+                "Program file to use. This is either an ELF shared-object file to be executed, \
+                 or an assembly file to be assembled and executed.",
+            )
+            .required(true)
+            .index(1);
         self.subcommand(
+            SubCommand::with_name("program")
+        .about("Run to test, debug, and analyze on-chain programs.")
+        .subcommand(
+            SubCommand::with_name("cfg")
+                .about("generates Control Flow Graph of the program.")
+                .arg(
+                    Arg::with_name("max_genesis_archive_unpacked_size")
+                        .long("max-genesis-archive-unpacked-size")
+                        .value_name("NUMBER")
+                        .takes_value(true)
+                        .default_value("10485760")
+                        .help("maximum total uncompressed size of unpacked genesis archive")
+                )
+                .arg(program_arg.clone())
+        )
+        .subcommand(
+            SubCommand::with_name("disassemble")
+                .about("dumps disassembled code of the program.")
+                .arg(
+                    Arg::with_name("max_genesis_archive_unpacked_size")
+                        .long("max-genesis-archive-unpacked-size")
+                        .value_name("NUMBER")
+                        .takes_value(true)
+                        .default_value("10485760")
+                        .help("maximum total uncompressed size of unpacked genesis archive")
+                )
+                .arg(program_arg.clone())
+        )
+        .subcommand(
             SubCommand::with_name("run")
-        .about(
-            r##"Run to test, debug, and analyze on-chain programs.
-
-The tool executes on-chain programs in a mocked environment.
+                .about(
+                    r##"The command executes on-chain programs in a mocked environment.
```

Comment:
I think we keep `The command executes on-chain programs in a mocked environment.` as the help message. However, I think we could move the block that explains how the JSON file needs to be formatted to the specific arg (`input` I believe).

At the moment, `solana-ledger-tool program --help` yields:
```
...
SUBCOMMANDS:
    cfg            generates Control Flow Graph of the program.
    disassemble    dumps disassembled code of the program.
    help           Prints this message or the help of the given subcommand(s)
    run            The command executes on-chain programs in a mocked environment.
                   
                   The input data for a program execution have to be in JSON format
                   and the following fields are required
                   {
                       "program_id": "DozgQiYtGbdyniV2T74xMdmjZJvYDzoRFFqw7UR5MwPK",
                       "accounts": [
                           {
                               "key": "524HMdYYBy6TAn4dK5vCcjiTmT2sxV6Xoue5EXrz22Ca",
                               "owner": "BPFLoaderUpgradeab1e11111111111111111111111",
                               "is_signer": false,
                               "is_writable": true,
                               "lamports": 1000,
                               "data": [0, 0, 0, 3]
                           }
                       ],
                       "instruction_data": [31, 32, 23, 24]
                   }
```
and I think it is only pertinent to show the input file message if the user is indeed going to move forward with `run` subcommand